Solvable example of an expanding beam scraping an aperture (#813)
BLAST-ImpactX/impactx: examples/scraping_beam/README.rst

.. _examples-scraping:

> Expanding Beam Scraping Against a Vacuum Pipe
> ==============================================

Reviewer: ```suggestion
Expanding Beam Scraping Against a Vacuum Pipe
=============================================
```

This example describes a coasting bunch, expanding transversely and encountering the aperture defined by the vacuum pipe.
Space charge is neglected, making the problem analytically soluble.

We use a cold (zero emittance) 250 MeV proton bunch whose

> initial distribution is a uniformly-populated cylinder of transverse radius rb = 2 mm with zero momentum spread.

Reviewer: ```suggestion
initial distribution is a uniformly-populated cylinder of transverse radius :math:`rb = 2 \mathrm{mm}` with zero momentum spread.
```

> The beam propagates in a drift with a vacuum chamber radius of R = 3.5 mm.

Reviewer: ```suggestion
The beam propagates in a drift with a vacuum chamber radius of :math:`R = 3.5 \mathrm{mm}`.
```

To generate an expanding beam, a linear map is first applied.  This map applies a radial kick to each particle that is proportional to the particle's initial distance from the axis.

> This induces a phase space correlation within the beam, such that px = k*x and py = k*y, similar to what would be induced by a space charge kick.

Reviewer: ```suggestion
This induces a phase space correlation within the beam, such that :math:`p_x = k \cdot x` and `p_y = k \cdot y`, similar to what would be induced by a space charge kick.
```

The beam remains cylindrical with zero emittance during its evolution in a 6 m drift.

> In the absence of an aperture, the beam radius evolves as :math:`rb(s) = rb(1 + ks)`.

Reviewer: If you like, you can use `r_b` and `k_s` as usual for subscripts.

> In the absence of an aperture, the beam radius evolves as :math:`rb(s) = rb(1 + ks)`.

Author: ```suggestion
In the absence of an aperture, the beam radius evolves as :math:`r_b(s) = r_b(1 + k\cdot s)`.
```

In the presence of an aperture, particles are lost during the transverse expansion.  The fraction of charge remaining after a distance s is given by:

> :math:`Q_s/Q_0 = \min{1,R^2/(rb^2(1+sk)^2)}`.

Reviewer: If you like, you can as in LaTeX also create math blocks:
```suggestion
.. math::

   Q_s/Q_0 = \min{1,R^2/(rb^2(1+sk)^2)}.
```
https://sphinx-rtd-trial.readthedocs.io/en/latest/ext/math.html

In this test, the initial and final values of :math:`\sigma_x`, :math:`\sigma_y`, :math:`\sigma_t`, :math:`\epsilon_x`, :math:`\epsilon_y`, and :math:`\epsilon_t` must agree with nominal values.

In addition, the initial and final values of :math:`\sigma_{p_x}`, :math:`\sigma_{p_y}`, and :math:`\sigma_{p_t}` must agree with nominal values.

Finally, the fraction of charge lost against the aperture at the exit of the drift must agree with nominal values.

> The physical problem is defined by four relevant parameters, defined within "run_scraping.py", that can be modified by the user:

Reviewer: Verbatim?
```suggestion
The physical problem is defined by four relevant parameters, defined within ``run_scraping.py``, that can be modified by the user:
```

> beam_radius = rb
> aperture_radius = R
> correlation_k = k
> drift_distance = s

Reviewer: Pretty syntax highlighting for the block:
```suggestion
.. code-block:: python

   # problem parameters
   beam_radius = rb
   aperture_radius = R
   correlation_k = k
   drift_distance = s
```

> These parameters should also be modified inside "analysis_scraping.py" for testing.

Reviewer: Verbatim?
```suggestion
These parameters should also be modified inside ``analysis_scraping.py`` for testing.
```

.. _examples-scraping:

Expanding Beam Scraping Against a Vacuum Pipe
=============================================

This example describes a coasting bunch, expanding transversely and encountering the aperture defined by the vacuum pipe.
Space charge is neglected, making the problem analytically soluble.

We use a cold (zero emittance) 250 MeV proton bunch whose

> initial distribution is a uniformly-populated cylinder of transverse radius :math:`rb = 2 \mathrm{mm}` with zero momentum spread.

Reviewer: ```suggestion
initial distribution is a uniformly-populated cylinder of transverse radius :math:`r_b = 2 \mathrm{mm}` with zero momentum spread.
```

The beam propagates in a drift with a vacuum chamber radius of :math:`R = 3.5 \mathrm{mm}`.

To generate an expanding beam, a linear map is first applied.  This map applies a radial kick to each particle that is proportional to the particle's initial distance from the axis.

> This induces a phase space correlation within the beam, such that :math:`p_x = k \cdot x` and `p_y = k \cdot y`, similar to what would be induced by a space charge kick.

Reviewer: ```suggestion
This induces a phase space correlation within the beam, such that :math:`p_x = k \cdot x` and :math:`p_y = k \cdot y`, similar to what would be induced by a space charge kick.
```

The beam remains cylindrical with zero emittance during its evolution in a 6 m drift.
In the absence of an aperture, the beam radius evolves as :math:`r_b(s) = r_b(1 + k\cdot s)`.
In the presence of an aperture, particles are lost during the transverse expansion.  The fraction of charge remaining after a distance s is given by:

.. math::

>    Q_s/Q_0 = \min{1,R^2/(r_b^2(1+s\cdot k)^2)}.

Reviewer: ```suggestion
   Q_s/Q_0 = \min\left{1,R^2/(r_b^2(1+s\cdot k)^2)\right}.
```

>    Q_s/Q_0 = \min\left{1,R^2/(r_b^2(1+s\cdot k)^2)\right}.

Reviewer: ```suggestion
   \frac{Q_s}{Q_0} = \min\left[1,R^2/(r_b^2(1+s\cdot k)^2)\right].
```

In this test, the initial and final values of :math:`\sigma_x`, :math:`\sigma_y`, :math:`\sigma_t`, :math:`\epsilon_x`, :math:`\epsilon_y`, and :math:`\epsilon_t` must agree with nominal values.

In addition, the initial and final values of :math:`\sigma_{p_x}`, :math:`\sigma_{p_y}`, and :math:`\sigma_{p_t}` must agree with nominal values.

Finally, the fraction of charge lost against the aperture at the exit of the drift must agree with nominal values.

The physical problem is defined by four relevant parameters, defined within ``run_scraping.py``, that can be modified by the user:

.. code-block:: python

   # problem parameters

>    beam_radius = :math:`r_b`
>    aperture_radius = :math:`R`
>    correlation_k = :math:`k`
>    drift_distance = :math:`s`

Reviewer: Python code block:
```suggestion
   beam_radius = r_b
   aperture_radius = R
   correlation_k = k
   drift_distance = s
```
